llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -20279,56 +20306,51 @@ bool SLPVectorizerPass::vectorizeStores(
   // dependencies and no need to waste compile time to try to vectorize them.
   // - Try to vectorize the sequence {1, {1, 0}, {3, 2}}.
   auto FillStoresSet = [&](unsigned Idx, StoreInst *SI) {
-    for (std::pair<unsigned, StoreIndexToDistSet> &Set : SortedStores) {
+    for (RelatedStoreInsts &StoreSeq : SortedStores) {
       std::optional<int> Diff = getPointersDiff(
-          Stores[Set.first]->getValueOperand()->getType(),
-          Stores[Set.first]->getPointerOperand(),
+          Stores[StoreSeq.BaseInstrIdx]->getValueOperand()->getType(),
+          Stores[StoreSeq.BaseInstrIdx]->getPointerOperand(),
           SI->getValueOperand()->getType(), SI->getPointerOperand(), *DL, *SE,
           /*StrictCheck=*/true);
       if (!Diff)
         continue;
-      auto It = Set.second.find(std::make_pair(Idx, *Diff));
-      if (It == Set.second.end()) {
-        Set.second.emplace(Idx, *Diff);
+      std::optional<unsigned> PrevInst = StoreSeq.getInstIdx(/*PtrDist=*/*Diff);
+      if (!PrevInst) {
+        StoreSeq.insert(Idx, *Diff);
```

Comment:
And update the call like so:
```suggestion
      std::optional<unsigned> PrevInst = StoreSeq.getOrInsertInstIdx(/*PtrDist=*/Diff, /*OrInsert*/);
      if (!PrevInst) {
```